web/src/main/ui/app/components/flows-new/edit-flow/ui/new-step-dialog-ui.component.ts

```diff
@@ -22,12 +19,14 @@ export class NewStepDialogUiComponent implements OnInit {
   @Input() collections: any;
   @Input() step: any;
   @Input() flow: Flow;
+  @Input() projectDirectory: string;
   @Output() getCollections = new EventEmitter();
   @Output() cancelClicked = new EventEmitter();
   @Output() saveClicked = new EventEmitter();
 
-  public newStep: Step = new Step;
-  readonly stepOptions = ['ingest', 'mapping', 'mastering', 'custom'];
+  public newStep: Step;
+  public stepType: typeof StepType = StepType;
```

Comment:
You don't need to instantiate the enum to use it. StepType.<whatever> should work.

Reply:
If I don't, it doesn't work on the HTML templating side

Reply:
Hm. Yeah. My bad. The template is translated and run within the execution of its component. Technically it does not see a global context. Thanks.